src/lib.rs

```diff
@@ -4221,6 +4255,291 @@ fn map_darwin_target_from_rust_to_compiler_architecture(target: &Target) -> &str
     }
 }
 
+#[derive(Debug)]
+struct RustcCodegenFlags {
+    branch_protection: Option<String>,
+    code_model: Option<String>,
+    no_vectorize_loops: bool,
+    no_vectorize_slp: bool,
+    profile_generate: Option<String>,
+    profile_use: Option<String>,
+    control_flow_guard: Option<String>,
+    lto: Option<String>,
+    relocation_model: Option<String>,
+    embed_bitcode: Option<bool>,
+    force_frame_pointers: Option<bool>,
+    link_dead_code: Option<bool>,
+    no_redzone: Option<bool>,
+    soft_float: Option<bool>,
+}
+
+impl RustcCodegenFlags {
+    // Parse flags obtained from CARGO_ENCODED_RUSTFLAGS
+    fn parse(rustflags_env: &str) -> RustcCodegenFlags {
```

Comment:
Nit: A few unit tests of this would be nice. Something like:

```rust
#[cfg(test)]
mod tests {
    use super::*;

    #[track_caller]
    fn check(env: &str, expected: &RustcCodegenFlags) {
        let actual = RustcCodegenFlags::parse(env);
        assert_eq!(actual, *expected);
    }

    #[test]
    fn codegen_type() {
        let expected = RustcCodegenFlags {
            code_model: Some("tiny".into()),
            ..RustcCodegenFlags::default()
        };
        check("-Ccode-model=tiny", &expected);
        check("-C\u{f1}code-model=tiny", &expected);
        check("--codegen\u{f1}code-model=tiny", &expected);
        check("--codegen=code-model=tiny", &expected);
    }

    #[test]
    fn precedence() {
        check("-Ccode-model=tiny\u{1f}-Ccode-model=small", RustcCodegenFlags {
            code_model: Some("small".into()),
            ..RustcCodegenFlags::default()
        });
    }

    // ...
}
```

Reply:
Maybe also a check for every single flag that `rustc` has:
```rust
let flags = [
    "--cfg",
    "a",
    "-L",
    "/usr/lib/foo",
    "-l",
    "static:+whole-archive=mylib",
    "--crate-type=dylib",
    // ...
    "-g",
    "-O",
    // ...
    "-Car",
    "-Ccode-model=tiny",
    "-Ccodegen-units=1",
    "-Ccollapse-macro-debuginfo=yes",
    // ...
];
check(flags.join('\u{f1}'), ...);
```

Reply:
Okay that's a hilarious edge-case, nice one. I think if we add -L as a prefix then this example would get joined to `-L-Clto` and then ignored because it doesn't match anything which would get the behaviour we want. How does that sound?

Reply:
Great idea!

I'd also prefer it for a lot of the other short flags (`-l`, `-o`, `-W`, `-A`, `-D`, `-F`), just be sure.